core/invoker/src/main/scala/whisk/core/invoker/InvokerReactive.scala

```diff
@@ -101,15 +102,16 @@ class InvokerReactive(
   private val topic = s"invoker${instance.toInt}"
   private val maximumContainers = (poolConfig.userMemory / MemoryLimit.minMemory).toInt
   private val msgProvider = SpiLoader.get[MessagingProvider]
-  private val consumer = msgProvider.getConsumer(
-    config,
-    topic,
-    topic,
-    maximumContainers,
-    maxPollInterval = TimeLimit.MAX_DURATION + 1.minute)
+
+  //number of peeked messages - increasing the concurrentPeekFactor improves concurrent usage, but adds risk for message loss in case of crash
+  private val maxPeek =
+    math.max(maximumContainers, (maximumContainers * limitsConfig.max * poolConfig.concurrentPeekFactor).toInt)
```

Comment:
I didn't quite follow your math, is there some typos? In any case, this will vary based on load, but the defaults should be better. 
I will change default max to 1 (concurrency disabled by default; only enabled by default in "local" env for test runs)

Reply:
Let me try again

```scala
val maximumContainers = (poolConfig.userMemory / MemoryLimit.minMemory).toInt
```

Here
* `MemoryLimit.minMemory` = 128 MB
* `poolConfig.userMemory` = x MB

So `maximumContainers` = x/128

Now
```scala
val maxPeek =
    math.max(maximumContainers, (maximumContainers * limitsConfig.max * poolConfig.concurrentPeekFactor).toInt)
```

Where earlier the defauls were
* `limitsConfig.max` = 500
* `poolConfig.concurrentPeekFactor` = 0.5

So 
```
maxPeek = max(x/128, x/128 * 500 * 0.5)
maxPeek = max(128x, ~ 2x)
```

So if `poolConfig.userMemory` is 32 GB = 32768 MB then `maxPeek` = 65536. Which is a high number. Considering a worst case of 1 MB per activation it results in loading 65 GB of data. So may be we introduce an explicit config for `maxPeek`?

Reply:
Would it be better to just define a different value for concurrentPeekFactor? 

My point is that regardless of specifying directly a value for maxPeek, or indirectly a value based on existing concurrentPeekFactor, there is still going to be potential for exhausting the system. e.g. we could set it to 0.1 - if you think that specifying maxPeek=12800 is easier to understand than concurrentPeekFactor=0.1, we can do that. But in neither case does it directly translate to actual behavior in the system since the message peeking a) pays no attention to differences in actions or activations request/response memory impact and b) is required to be increased to allow concurrent activations to be processed. This is a known tradeoff of enabling concurrency for now.